### ISSUE TITLE
Missing in props validation when passing data to children component

### ISSUE THREAD

**Issue by xihaoxiaohao** [yannickcr/eslint-plugin-react#1615]
I try to pass the `this.state.videos` to my video_list.js file, but in my video_list.js file, eslint is showing these errors:
"videos is missing props validation (react/prop-types)"; 
"videos.length is missing props validation (react/prop-types)".

In my index.js file:

```
class App extends Component {
    constructor(props) {
        super(props);
        this.state = { videos: [] };
        YTSearch({ key: API_KEY, term: 'Surfboards' }, videos => {
            this.setState({ videos });
        });
    }

    render() {
        return (
            <div>
                <SearchBar />
                <VideoList videos={this.state.videos} />
            </div>
        );
    }
}
```

video_list.js file:

```
import React from 'react';

const VideoList = (props) => {
    return <ul className="col-md-4" list-group>
        {props.videos.length}
    </ul>;
};

export default VideoList;
```

How can I fix the errors or how to modify the settings to ignore it? #

**Comment by ljharb** [comment]
The solution is to add propTypes to VideoList:
```js
import PropTypes from 'prop-types';

VideoList.propTypes = {
  videos: PropTypes.arrayOf(PropTypes.shape(…)); // whatever `this.state.videos` is
};
```

**Comment by istana** [comment]
I just stumbled upon this issue from web search and I think it can be closed since months passed without any response.
